src/components/Game/Game.tsx

```diff
@@ -27,33 +28,85 @@ type GameProps = {
   game: Game;
 };
 
+const levelNames = [
+  'Ameise',
+  'Regenwurm',
+  'Laubfrosch',
+  'Salamander',
+  'Hase',
+  'Fuchs',
+  'Waldschrat',
+  'Bär',
+  'Waldfee',
+];
+
 export const Game = ({ game }: GameProps) => {
-  const { letters, matchedWords } = game;
+  const { letters, totalScore, /* levelScores, */ matchedWords, panagrams } = game;
+  const levelScores = [0, 7, 19, 33, 48, 64, 80, 97, 114, 190];
+
   const [gameLetters, setGameLetters] = useState(letters);
   const [wordInput, setWordInput] = useState('');
   const [foundWords, setFoundWords] = useState<string[]>([]);
   const [isAnimation, setIsAnimation] = useState(false);
+  const [curScore, setCurScore] = useState(0);
+  const levels = levelScores.map((score, i) => ({ level: i + 1, levelName: levelNames[i], score }));
+  const [curLevel, setCurLevel] = useState(levels[0]);
```

Comment:
I would go for not setting the level in state. We can derive the state from the currentScore.
This removes the need for a useEffect.
For that we find the levelObject whose score is smaller or equal to the curScore, but the following object's score is not smaller or equal to the curScore.
Also with this, we can put this calculation directly into the level component, as this information is not needed anywhere else.